core/domain/user_jobs_one_off.py

```diff
@@ -68,6 +68,29 @@ def reduce(key, version_and_exp_ids):
                     edited_exploration_ids))
 
 
+class UsernameLengthDistributionOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for calculating the distribution of username lengths."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(item):
+        if item.username is not None:
+            yield (len(item.username), {
+                'username': item.username
+            })
+
+    @staticmethod
+    def reduce(key, usernames):
+        username_list = [ast.literal_eval(v) for v in usernames]
+        username_length_model = (
```

Comment:
I don't think you need a new model to store this. Just yield the summary. It will show up in the admin page after running the job.

See e.g. InteractionAuditOneOffJob in core/domain/exp_jobs_one_off.py.